src/main/java/org/gooru/insights/services/BaseESServiceImpl.java

```diff
@@ -414,21 +426,16 @@ private void buildBuckets(String traceId,String index, RequestParamsDTO requestP
 					termBuilder = AggregationBuilders.terms(groupBy[i]).field(fieldName);
 				}
 				if (i == groupBy.length - 1) {
-					bucketAggregation(index, requestParamsDTO, termBuilder, metricsName);
+					includeMetricsAggregation(index, requestParamsDTO, filterAggregationBuilder, searchRequestBuilder, termBuilder, null, null, metricsName);
```

Comment:
includeMetricsAggregation(index, requestParamsDTO, filterAggregationBuilder, searchRequestBuilder, termBuilder, null, null, metricsName);

there might be chance of NPE when filterAggregationBuilder is null